File: TTTapAndHoldMenu/Core/PinterestLikeMenu/PinterestLikeMenuItem.m
//
//  PininterestLikeMenuItem.m
//  PininterestLikeMenu
//
//  Created by Tu You on 12/21/13.
//  Copyright (c) 2013 Tu You. All rights reserved.
//

#import "PinterestLikeMenuItem.h"



@interface PinterestLikeMenuItem ()

@property (nonatomic, strong) UIImageView *imageView;

@end


@implementation PinterestLikeMenuItem

- (id)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        // Initialization code
    }
    return self;
}

- (id)initWithImage:(UIImage *)image selctedImage:(UIImage *)selectedImage selectedBlock:(SelectedBlock)selectedBlock
{
    if (self = [super init])
    {
        self.bounds = CGRectMake(0, 0, kMenuItemLength, kMenuItemLength);
        
        self.imageView = [[UIImageView alloc] initWithImage:image];
        self.imageView.frame = self.bounds;
        self.imageView.image = image;
        self.imageView.highlightedImage = selectedImage;
        
        self.label = [UILabel new];
        
        [self addSubview:self.imageView];
        
        self.selectedBlock = selectedBlock;
        
    }
    return self;
}

- (void)setSelected:(BOOL)selected
{
    _selected = selected;
    
    self.imageView.highlighted = selected;
}


@end

File: TTTapAndHoldMenu/Core/PinterestLikeMenu/PinterestLikeMenuItem.swift
//
//  PinterestLikeMenuItem.swift
//  TTTapAndHoldMenu
//
//  Created by alysenko on 09/03/16.
//  Copyright © 2016 alysenko. All rights reserved.
//

import Foundation

class PinterestLikeMenuItem : UIView {
    
    var image: UIImage {
        didSet {
            imageView.image = image
        }
    }
    var selectedImage: UIImage {
        didSet {
            imageView.highlightedImage = selectedImage
        }
    }
    
    var selectedBlock: (Void -> Void)
    
    var label: UILabel = UILabel()
    
    var selected: Bool = false {
        didSet {
            imageView.highlighted = selected
        }
    }
    
    var distance: CGFloat = 0
    
    private var imageView: UIImageView = UIImageView()
    
    var imageSize = CGSizeMake(40, 40)
    var selectedImageSize = CGSizeMake(50, 50)
    
    var currentImageSize = CGSizeMake(40, 40) {
        didSet {
            imageView.Size = currentImageSize
        }
    }
    
    init(image: UIImage, imageSize: CGSize, selectedImage: UIImage, selectedImageSize: CGSize, selectedBlock: (Void -> Void)) {
        self.image = image
        self.selectedImage = selectedImage
        
        self.imageSize = imageSize
        self.selectedImageSize = selectedImageSize
        
        self.selectedBlock = selectedBlock
        
        let width = max(imageSize.width, selectedImageSize.width)
        let height = max(imageSize.height, selectedImageSize.height)
        super.init(frame: CGRectMake(0, 0, width, height))
        
        self.addSubview(imageView)
        self.imageView.frame = self.bounds
        
        currentImageSize = imageSize
        imageView.Size = currentImageSize
        
        imageView.image = image
        imageView.highlightedImage = selectedImage
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}